Formula/udp2raw-multiplatform.rb

```diff
@@ -0,0 +1,30 @@
+class Udp2rawMultiplatform < Formula
+  desc "Multi-platform(cross-platform) version of udp2raw-tunnel client"
+  homepage "https://github.com/wangyu-/udp2raw-multiplatform"
+  url "https://github.com/wangyu-/udp2raw-multiplatform/archive/refs/tags/20210111.0.tar.gz"
+  sha256 "712ad3c79b6ef5bf106c615823d0b0b3865d1c957f9838cf05c23b7ac7024438"
+  license "MIT"
+
+  depends_on "make" => :build
+  depends_on "libnet"
+  depends_on :macos
+
+  def install
+    ENV["OPT"] = (system_command "libnet-config", args: ["--cflags"]).stdout
+    system "make", "mac"
+    bin.install "udp2raw_mp"
+    etc.install "example.conf" => "udp2raw_client.conf"
+  end
+
+  service do
+    run [opt_bin/"udp2raw_mp", "--conf-file", etc/"udp2raw_client.conf"]
+    keep_alive true
+    require_root true
+    log_path var/"log/udp2raw.log"
+    error_log_path var/"log/udp2raw.log"
+  end
+
+  test do
+    assert_match "udp2raw-tunnel", shell_output("#{bin}/udp2raw_mp -h")
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
udp2raw-multiplatform is a client only implementation of the original udp2raw for better multi-platform support. There is no way to properly test the client without launching a server first.

Reply:
In the link it specifies:
> If the software cannot function without credentials or requires a virtual machine, docker instance, etc. to run, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected. This is preferred over mocking a dependency.

Which I think can easily apply here.

Reply:
New test by checking the permission denied failure.